agency/general-agent.go

```diff
@@ -141,6 +144,7 @@ func (agentState *GeneralAgentInfo) getSystemMessage() (*engines.Message, error)
 		Content: contextString,
 		ID:      &messageId,
 	}
+	agentState.resultsProcessingChannel <- systemMessage
```

Comment:
what is the purpose of sending `systemMessage` into the `resultsProcessingChannel`? why don't you want to send it straight to the history appender channel?

Reply:
You are right, it should be sent straight to the other channel. Was simply copying the channel send from the preceding function "agentStateResultsReceiver" but did not quite realize at the time that this was unnecessary for the systemMessage purpose.